### PR TITLE
[ID-79]Consolidate the information, and make it accessible with a single API call

### LINKED ISSUE #79: [FEATURE] Consolidate the information, and make it accessible with a single API call
**Is your feature request related to a problem? Please describe.**
API that returns participant's data and all surveys or survey responses they have participated in.

### PR DESCRIPTION
**Pull request by stefanvit**
## Description
Consolidate the information, and make it accessible with a single API call
**Resolves #79 

## Review Time Estimate
- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [x] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by stefanvit** [comment]
Hi @petyos, before reviewing, please only generate the docs. Thank you.

File: core/app_shared.go [hunk 1]
```diff
@@ -152,6 +152,37 @@ func (a appShared) hasAttendedEvent(orgID string, appID string, eventID string,
 	return false, nil
 }
 
+func (a appShared) getUserData(orgID string, appID string, userID *string) (*model.UserData, error) {
```
Comment (petyos) [inline core/app_shared.go:155]: As we can have more data in the future as general the approach of getting this info is to become asynchronously. Once all data has been loaded then you are ready to return the result.

**Review by petyos — CHANGES_REQUESTED** [review]
Please look at the comment.

**Review by petyos — APPROVED** [review]
LGTM

**Pull request merged**